homeassistant/components/group/media_player.py

```diff
@@ -500,5 +537,13 @@ def async_update_group_state(self) -> None:
             supported_features |= MediaPlayerEntityFeature.MEDIA_ANNOUNCE
         if self._features[KEY_ENQUEUE]:
             supported_features |= MediaPlayerEntityFeature.MEDIA_ENQUEUE
+        if self._features[KEY_SELECT_SOURCE]:
+            supported_features |= MediaPlayerEntityFeature.SELECT_SOURCE
 
         self._attr_supported_features = supported_features
+        if len(self._entity_sources):
+            self._attr_source_list = list(
+                set.intersection(*self._entity_sources.values())
```

Comment:
I think this may need an architecture discussion to make sure we have thought about and agreed how the state should be updated.

Please review other group platforms, how other state attributes are updated and present this in the post and motivate why your suggested solution is appropriate.

https://github.com/home-assistant/architecture/discussions/

Reply:
Created one https://github.com/home-assistant/architecture/discussions/1209

Reply:
so discussion is looking like a no for the source_list If the PR is trimmed to just the SELECT_SOURCE service call would that be acceptable

Reply:
@MartinHjelmare ?